File: Chapter 2-9-Detect Target Running/Objective-C/DetectTarget/DetectTarget/ViewController.m
//
//  ViewController.m
//  DetectTarget
//
//  Copyright © 2015 - by Magno Urbano - All rights reserved.
//  Part of the book Troubleshooting Xcode, published by Apress
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
  
  NSLog(@"Target Running = %@", [self getTarget]);
}


- (NSString *) getTarget {
  
  NSDictionary *dict = [[NSBundle mainBundle] infoDictionary];
  NSArray *family = dict[@"UIDeviceFamily"];
  NSInteger device = [family[0] integerValue];
  
  return (device == 1) ? @"iPhone" : @"iPad";
}





- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

@end

File: Chapter 2-9-Detect Target Running/Swift 2.0/DetectTarget/DetectTarget/ViewController.swift
//
//  ViewController.swift
//  DetectTarget
//
//  Copyright © 2015 - by Magno Urbano - All rights reserved.
//  Part of the book Troubleshooting Xcode, published by Apress
//

import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view, typically from a nib.
    
    print("Target Running = ",  getTarget());
  }

  
  func getTarget() -> NSString {
    
    let dict : NSDictionary = NSBundle.mainBundle().infoDictionary!
    
    let family  = dict["UIDeviceFamily"]!
    let device = family[0]
    
    return (device.integerValue == 1) ? "iPhone" : "iPad"
  }

  
  
  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }


}

